Sources/Services/AWSRoute53/Route53Client.swift

```diff
@@ -159,14 +159,15 @@ extension Route53Client {
                       .withAuthSchemes(value: config.authSchemes ?? [])
                       .withAuthSchemeResolver(value: config.authSchemeResolver)
                       .withUnsignedPayloadTrait(value: false)
+                      .withSocketTimeout(value: config.httpClientConfiguration.socketTimeout)
                       .withIdentityResolver(value: config.awsCredentialIdentityResolver, schemeID: "aws.auth#sigv4")
                       .withIdentityResolver(value: config.awsCredentialIdentityResolver, schemeID: "aws.auth#sigv4a")
                       .withRegion(value: config.region)
                       .withSigningName(value: "route53")
                       .withSigningRegion(value: config.signingRegion)
                       .build()
         var operation = ClientRuntime.OperationStack<ActivateKeySigningKeyInput, ActivateKeySigningKeyOutput>(id: "activateKeySigningKey")
-        operation.initializeStep.intercept(position: .after, middleware: ActivateKeySigningKeyInputStripHostedZoneMiddleware())
+        operation.initializeStep.intercept(position: .after, middleware: Route53TrimHostedZoneMiddleware(\.hostedZoneId))
```

Comment:
Middleware insertion is updated to use the new, reusable middleware and to pass in the key path to the field holding the hosted zone ID.

Six additional operations get the customization as well; these are ones that use a hosted zone ID but it is not exactly named "HostedZoneId".